STL_Extension/include/CGAL/Small_stack_allocator.h

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2020  GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Simon Giraudot
+
+#ifndef CGAL_SMALL_STACK_ALLOCATOR_H
+#define CGAL_SMALL_STACK_ALLOCATOR_H
+
+#include <CGAL/Profile_counter.h>
+
+#include <cstdlib>
+
+namespace CGAL
+{
+
+namespace internal
+{
+
+template <typename T, std::size_t MaxSize>
+class Small_stack_allocator_pool
+{
+  T m_pool[MaxSize];
+  std::size_t m_next;
+
+public:
+
+  Small_stack_allocator_pool ()
+    : m_next(0) { }
+  ~Small_stack_allocator_pool() { }
+
+  // Forbid copy/assignment
+  Small_stack_allocator_pool (const Small_stack_allocator_pool&) = delete;
+  Small_stack_allocator_pool& operator= (const Small_stack_allocator_pool&) = delete;
+
+  T* allocate (std::size_t n)
+  {
+    CGAL_BRANCH_PROFILER("stack allocations / total allocations of Small_stack_allocator", prof);
+
+    // If chunk to allocate does not exceed remaining stack size, use stack
+    if (m_next + n < MaxSize)
+    {
+      CGAL_BRANCH_PROFILER_BRANCH(prof);
+      T* out = m_pool + m_next;
+      m_next += n;
+      return out;
+    }
+
+    // Else, fallback to `new` allocation
+    return new T[n];
+  }
+
+  void deallocate (T* p, std::size_t n)
+  {
+    // If pointer is part of the pool, nothing to do
+    if (m_pool <= p && p < m_pool + MaxSize)
```

Comment:
The legality of this kind of test regularly comes up...

Reply:
What would be the alternative?